utils.go

```diff
@@ -407,6 +407,39 @@ func ReadStringWithMax(r io.Reader, maxLength uint64) (string, error) {
 	return string(buf), nil
 }
 
+// ReadFullStringIntoBuf will read a string off the given stream, consuming the
+// entire cbor item if the string on the stream is longer than the buffer,
+// the string is discarded and 'false' is returned
+func ReadFullStringIntoBuf(cr *CborReader, buf []byte, maxLength uint64) (int, bool, error) {
```

Comment:
nit: generally I prefer the `Hash` version of this:

1. Take a zero-length buffer with some capacity.
2. Return a buffer containing the output, appending to the input buffer if possible.

That way it's possible to just pass `nil` and let the function do the allocation for you. You also won't need to handle slicing, etc.

Reply:
So the main reason i added this was to avoid any allocations, i can make it append if necessary but it feels more prone to being misused in that case

Reply:
Ah, I see. You want to be able to skip fields that are too long entirely. I guess that makes sense, it just seems like a bit of a sharp edge.

Reply:
ill add an extra comment